sdk/javascript/src/RuleBasedTransactionFactory.js

```diff
@@ -40,8 +40,11 @@ const typeConverterFactory = (module, customTypeConverter, value) => {
 	if (customTypeConverter && customTypeConverter(value))
 		return customTypeConverter(value);
 
-	if (value instanceof ByteArray)
-		return new module[value.constructor.name](value.bytes);
+	if (value instanceof ByteArray) {
+		/** @type object */
```

Comment:
since this is in `instanceof` block, it must be a `ByteArray` type with NAME static property

but, TS still detects `value.constructor` as `function` (not `ByteArray`) and disallows assignment because `function` may not have `NAME`.